elide-datastore/elide-datastore-jms/src/main/java/com/yahoo/elide/datastores/jms/websocket/SubscriptionWebSocketConfigurator.java

```diff
@@ -81,22 +89,68 @@ public class SubscriptionWebSocketConfigurator extends ServerEndpointConfig.Conf
     @Builder.Default
     protected DataFetcherExceptionHandler dataFetcherExceptionHandler = new SimpleDataFetcherExceptionHandler();
 
+    @Builder.Default
+    protected RouteResolver routeResolver = null;
+
+    @Builder.Default
+    protected Injector injector = null;
+
     @Override
     public <T> T getEndpointInstance(Class<T> endpointClass) throws InstantiationException {
         if (endpointClass.equals(SubscriptionWebSocket.class)) {
 
-            EntityDictionary dictionary = EntityDictionary.builder().build();
+            EntityDictionary dictionary = EntityDictionary.builder().injector(injector).build();
 
             DataStore store = buildDataStore(dictionary);
 
             Elide elide = buildElide(store, dictionary);
 
-            return (T) buildWebSocket(elide);
+            return endpointClass.cast(buildWebSocket(elide));
         }
 
         return super.getEndpointInstance(endpointClass);
     }
 
+    @Override
+    public void modifyHandshake(ServerEndpointConfig sec, HandshakeRequest request, HandshakeResponse response) {
+        Map<String, Object> userProperties = sec.getUserProperties();
+        URI requestURI = request.getRequestURI();
+        if (requestURI != null)
+        {
+            userProperties.put("requestURI", requestURI);
+        }
+
+        Map<String, List<String>> headers = request.getHeaders();
+        if (headers != null)
+        {
+            userProperties.put("headers", headers);
+        }
+
+        Map<String, List<String>> parameterMap = request.getParameterMap();
+        if (parameterMap != null)
+        {
+            userProperties.put("parameterMap", parameterMap);
+        }
+
+        String queryString = request.getQueryString();
+        if (queryString != null)
+        {
+            userProperties.put("queryString", queryString);
+        }
+
+        Object httpSession = request.getHttpSession();
+        if (httpSession != null)
+        {
+            userProperties.put("session", httpSession);
+        }
+
+        Principal userPrincipal = request.getUserPrincipal();
+        if (userPrincipal != null)
+        {
+            userProperties.put("userPrincipal", userPrincipal);
```

Comment:
Is there some plan later to potentially use this principal to construct the Elide user object?

Reply:
This user principal here should be the same one that can be accessed from the `jakarta.websocket.Session.getUserPrincipal` method.

The default `SubscriptionWebSocket.UserFactory` already creates a user using `session -> new User(session.getUserPrincipal());`